src/com/mendix/widget/GoogleMaps/components/__tests__/Map.spec.ts

```diff
@@ -0,0 +1,201 @@
+                    bootstrapURLKeys: { key: undefined },
+                    center: defaultCenterLocation,
+                    defaultZoom: 14,
+                    onGoogleApiLoaded: jasmine.any(Function) as any,
+                    resetBoundsOnResize: true,
+                    yesIWantToUseGoogleMapApiInternals: true
+                })
+            )
+        );
+    });
+
+    it("renders with classes", () => {
+        const googleMap = renderMap({ address });
+
+        expect(googleMap.hasClass("widget-google-maps")).toBe(true);
+    });
+
+    describe("with no address", () => {
+        it("should not look up the location", () => {
+            spyOn(window.google.maps.Geocoder.prototype, "geocode").and.callThrough();
+
+            const output = renderMap({ address: "" });
+            output.find(GoogleMap).prop("onGoogleApiLoaded")();
+
+            expect(window.google.maps.Geocoder.prototype.geocode).not.toHaveBeenCalled();
+        });
+
+        it("should not display a marker", () => {
+            const output = renderMap({ address: "" });
+            output.find(GoogleMap).prop("onGoogleApiLoaded")();
+
+            const marker = output.find(Marker);
+            expect(marker.length).toBe(0);
+        });
+
+        it("should center to the default address", () => {
+            const googleMap = renderMap({ address }).childAt(0);
+
+            expect(googleMap.prop("center").lat).toBe(defaultCenterLocation.lat);
+            expect(googleMap.prop("center").lng).toBe(defaultCenterLocation.lng);
+        });
+    });
+
+    describe("with a valid address", () => {
+        it("should lookup the location", () => {
+            spyOn(window.google.maps.Geocoder.prototype, "geocode");
+
+            const output = renderMap({ address });
+            output.find(GoogleMap).prop("onGoogleApiLoaded")();
+
+            expect(window.google.maps.Geocoder.prototype.geocode).toHaveBeenCalled();
+        });
+
+        it("should render a marker", () => {
+            const output = renderMap({ address });
+            output.find(GoogleMap).prop("onGoogleApiLoaded")();
+
+            const marker = output.find(Marker);
+            expect(marker.length).toBe(1);
+            expect(marker.prop("lat")).toBe(successMockLocation.lat);
+            expect(marker.prop("lng")).toBe(successMockLocation.lng);
+        });
+
+        it("should center to the location of the address", () => {
+            const output = renderMap({ address });
+            const mapComponent = output.instance() as Map;
+            output.find(GoogleMap).prop("onGoogleApiLoaded")();
+
+            expect(mapComponent.state.location.lat).toBe(successMockLocation.lat);
+            expect(mapComponent.state.location.lng).toBe(successMockLocation.lng);
+        });
+
+        it("should display the first marker if multiple locations are found", () => {
+            const output = renderMap({ address: "multipleAddress" });
```

Comment:
Code for setting up the mock should move into the same file. Otherwise you have to check multiple files to understand what exactly test is doing.